database/ffldb/blockio.go

```diff
@@ -553,6 +575,38 @@ func (s *blockStore) readBlock(hash *chainhash.Hash, loc blockLocation) ([]byte,
 	return serializedData[8 : n-4], nil
 }
 
+// deleteBlocks will delete all blocks in the blockstore where the height of the
+// last block in the file is less than the provided height.
+func (s *blockStore) deleteBlocks(deleteBefore uint32) error {
+	var toDelete []uint32
```

Comment:
I'd move this after of the RLock is taken and do `toDelete := make([]uint32, 0, len(s.fileBlockHeights)`